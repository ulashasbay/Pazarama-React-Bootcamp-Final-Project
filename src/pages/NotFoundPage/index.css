#mainC {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  color: #000;
  font-family: sans-serif;
  max-width: 340px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -o-user-select: none;
  user-select: none;
}

.message {
  font-size: 16px;
  text-align: center;
}

.message h1 {
  margin: 0;
  padding: 0;
  font-size: 11em;
  transform: skewY(-5deg);
  transition: 0.4s ease-in-out all;
}

.message h3 {
  color: #000;
  font-size: 0.9em;
  font-weight: lighter;
  line-height: 1;
  margin-bottom: 20px;
}

.footer {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  width: 100%;
}

.footer a {
  position: relative;
  flex: 1;
  background-color: #0137f3;
  color: #fff;
  text-decoration: none;
  border-radius: 3px;
  border: 1px solid #0137f3;
  margin: 0 20px;
  text-align: center;
  padding: 10px 0;
  overflow: hidden;
  transition: all 0.2s ease-in-out;
  z-index: 3;
  cursor: pointer;
}

.footer a:hover {
  border: 1px solid rgba(255, 255, 255, 1);
  color: #fff;
  background-color: #365deb ;
}
