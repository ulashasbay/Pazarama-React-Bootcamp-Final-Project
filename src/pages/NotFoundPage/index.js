import React from "react";
import { Link } from "react-router-dom";
import "./index.css";

function NotFoundPage() {
  return (
    <div id="mainC">
      <div className="message">
        <h1>404</h1>
        <h3>The page you seek does not exist</h3>
      </div>
      <div className="footer">
        <Link to="/">Go Home</Link>
      </div>
    </div>
  );
}

export default NotFoundPage;
