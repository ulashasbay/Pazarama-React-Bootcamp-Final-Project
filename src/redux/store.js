import { configureStore } from "@reduxjs/toolkit";
import appIdReducer from "./appId/appIdSlice";
import userInfoReducer from "./userInfo/userInfoSlice";
import isLoggedInReducer from "./isLoggedIn/isLoggedInSlice";

export const store = configureStore({
  reducer: {
    appId: appIdReducer,
    userInfo: userInfoReducer,
    isLoggedIn: isLoggedInReducer,
  },
});
