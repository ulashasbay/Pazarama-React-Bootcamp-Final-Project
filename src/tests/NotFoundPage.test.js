import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import NotFoundPage from "../pages/NotFoundPage";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { store } from "../redux/store";
import renderer from "react-test-renderer";

const AllTheProviders = ({ children }) => {
  return (
    <BrowserRouter>
      <Provider store={store}>{children}</Provider>
    </BrowserRouter>
  );
};

describe("NotFoundPage Tests", () => {
  beforeEach(() => {
    render(<NotFoundPage />, { wrapper: AllTheProviders });
  });

  test("Render header", () => {
    const header = screen.getByText("404");
    expect(header).toBeInTheDocument();
  });

  test("Click home button redirect to HomePage", () => {
    const button = screen.getByText("Go Home");
    userEvent.click(button);
    expect(location.pathname).toEqual("/");
  });
});

test("NotFoundPage renders correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Provider store={store}>
          <NotFoundPage />
        </Provider>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
