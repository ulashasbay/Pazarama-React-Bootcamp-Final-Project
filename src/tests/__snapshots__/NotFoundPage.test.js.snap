// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage renders correctly 1`] = `
<div
  id="mainC"
>
  <div
    className="message"
  >
    <h1>
      404
    </h1>
    <h3>
      The page you seek does not exist
    </h3>
  </div>
  <div
    className="footer"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Go Home
    </a>
  </div>
</div>
`;
